src/decoder.js

```diff
@@ -37,22 +37,48 @@ function decoder(mtype) {
 
         // Map fields
         if (field.map) { gen
-                ("r.skip().pos++") // assumes id 1 + key wireType
                 ("if(%s===util.emptyObject)", ref)
                     ("%s={}", ref)
-                ("k=r.%s()", field.keyType)
-                ("r.pos++"); // assumes id 2 + value wireType
-            if (types.long[field.keyType] !== undefined) {
-                if (types.basic[type] === undefined) gen
-                ("%s[typeof k===\"object\"?util.longToHash(k):k]=types[%i].decode(r,r.uint32())", ref, i); // can't be groups
-                else gen
-                ("%s[typeof k===\"object\"?util.longToHash(k):k]=r.%s()", ref, type);
-            } else {
-                if (types.basic[type] === undefined) gen
-                ("%s[k]=types[%i].decode(r,r.uint32())", ref, i); // can't be groups
-                else gen
-                ("%s[k]=r.%s()", ref, type);
-            }
+                ("var c2 = r.uint32()+r.pos");
+
+            if (field.keyType === "string") gen
+                ("k=\"%s\"", types.defaults[field.keyType]);
+            else if (types.defaults[field.keyType] !== undefined) gen
+                ("k=%s", types.defaults[field.keyType]);
+            else gen
+                ("k=null");
+
+            if (type === "string") gen
+                ("value=\"%s\"", types.defaults[type]);
+            else if (types.defaults[type] !== undefined) gen
+                ("value=%s", types.defaults[type]);
```

Comment:
if type === "bytes" then this tries to insert the default value of an empty array with %s and it ends up inserting empty string "" instead of "[]".  This is because %s calls String([]) when we need %j here so we use JSON.stringify([]).

Reply:
Thanks, fixed.